sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubConsumerAsyncClientIntegrationTest.java

```diff
@@ -441,7 +442,8 @@ public void receivesMultiplePartitions() {
         // Act & Assert
         try {
             StepVerifier.create(consumer.receive(false)
-                .filter(x -> TestUtils.isMatchingEvent(x.getData(), MESSAGE_TRACKING_ID))
+                .filter(x -> TestUtils.isMatchingEvent(x.getData(), MESSAGE_TRACKING_ID)
+                    && TestUtils.isMatchPartition(x.getPartitionContext().getPartitionId(), matchPartitions))
```

Comment:
There are two sets being mutated here `matchPartitions` and `expectedPartitions`. We don't need TestUtils.isMatchPartition either, that is asserted in assertPartitionEvent. 

You can update this to:

```java
&& expectedPartitions.contains(x.getPartitionContext().getPartitionId()))
```

since `expectedPartitions` is the set that we are still waiting on data from.

Reply:
Fixed in the new version.